### DISCUSSION TITLE
Help me to edit and improve a lua script (cut-video.lua)

### DISCUSSION THREAD

**Discussion by KonoVitoDa** [mpv-player/mpv#14215] [Q&A]
Hello everyone! I use [this script](https://github.com/samhippo/mpv-scripts/blob/d49768d9624d428ec158d21921a79e0b871ba9d6/cut-video.lua), which is a modded version of `cut-video.lua` I made; `cut-video.lua` is the best script to clip parts of videos I've found so far, and I want to improve it even more by adding some new features to it, but I'm too inexperienced with programming to do it by myself, so I would really appreciate if someone could help me. 🙏

These are the features I want to add:

* (Solved) Include miliseconds in `output_filename` (and show them into the [cut points panel](https://i.imgur.com/K56azI6.png));
* (Solved) Preserve `cut-start/end` points after executing `cut-finish` and only clear it when a keybind defined by user is pressed;
* Add keybinds to navigate through `cut-start/end` points;
* (Solved) Only includes the hour field *(HH:MM:SS.mss*) in output's filename if the file has more than a hour, and otherwise left it with only *MM:SS.mss*.

**Comment by avih** [comment]
There's no simple answer to your questions.

It's like asking "how do I add features A, B, and C to program FOO?" the answer is that you need to know or learn the programming language of FOO, and figure it out. If there was a simple answer then you wouldn't need to spend years to be a programmer.

To write or edit a script one needs to have reasonable experience with programming in general, and specifically with lua, and also know (or learn) how mpv lua scripts interact with mpv.

If you have a very specific questions, like "`mp.get_property("time-pos")` doesn't include milliseconds. how do I get the exact timestamp with milliseconds?" then you should first ask at the mpv IRC channel, and hopefully someone would be able to help.

But it's impossible to answer your questions without spending the time and doing the work for you, which people are unlikely to do.
